provider/resource_destination.go

```diff
@@ -116,3 +119,29 @@ func resourceDestinationRead(d *schema.ResourceData, meta any) error {
 func resourceDestinationDelete(d *schema.ResourceData, meta any) error {
 	return genericResourceDelete(model.KindDestination, d, meta)
 }
+
+func resourceDestinationImportState(_ context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	bindplane := meta.(*client.BindPlane)
+
+	// When importing, name is not set in the state so we need to grab
+	// the ID instead, which is the same as "name".
+	destName := d.Id()
```

Comment:
When importing, `d.Id()` is the name passed to the import command.

```
terraform import bindplane_destination.custom <id>
```

In the case of destinations, ID is always the same as name as far as the provider is concerned.